test/index.spec.js

```diff
@@ -269,26 +286,30 @@ describe('multihash', () => {
 
   describe('coerceCode', () => {
     it('valid', () => {
-      /** @type {Partial<Record<import('../src/constants').HashName, import('../src/constants').HashCode>> } */
+      /** @type {Partial<Record<HashName, HashCode>> } */
       const names = {
         sha1: 0x11,
         'sha2-256': 0x12,
         'sha2-512': 0x13,
         'sha3-512': 0x14
       }
 
-      Object.keys(names).forEach((name) => {
+      // eslint-disable-next-line guard-for-in
+      for (const name in names) {
         expect(
-          mh.coerceCode(/** @type {import('../src/constants.js').HashName} */(name))
+          mh.coerceCode(/** @type {keyof typeof names} */(name))
```

Comment:
`keyof typeof` is new to me - is this saying "look up the key type in the type declaration for `names`?

Reply:
`name` should be one of the keys of the `names` type

Reply:
Example might help here, e.g if `names` is `{x: 1, y: 2}` then `typeof name` is `{ x:number, y:number }` and `keyof typeof name` is type union `'x'|'y'` implying that both `'x'` and `'y'` would satisfy it.

Reply:
👍 thanks